### PR TITLE
[CHC] CA-366069 & XSI-1222: Fix incorrect capitalisation of value in Japanese version + add some case insensitive checks

### PR DESCRIPTION
**Pull request by danilo-delbusso**

### CODE REVIEW

File: XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectPatchPage.cs [hunk 1]
```diff
@@ -232,3 +232,4 @@ private void fileNameTextBox_TextChanged(object sender, EventArgs e)
 
     public enum UpdateType { Legacy, ISO }
 }
+
```
Comment (CitrixChris) [inline XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectPatchPage.cs:235]: Random new line at end of file

**Review by CitrixChris — CHANGES_REQUESTED** [review]

File: XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectPatchPage.cs (hunk 1 above)
Comment (danilo-delbusso) [reply XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectPatchPage.cs]: ✔️

**Review by CitrixChris — APPROVED** [review]

**Review by AaronRobson — APPROVED** [review]

**Review by kc284 — CHANGES_REQUESTED** [review]

File: XenModel/Actions/Updates/DownloadAndUnzipXenServerPatchAction.cs [hunk 2]
```diff
@@ -200,7 +200,7 @@ private void ExtractFile()
 
                     while (iterator.HasNext())
                     {
-                        string currentExtension = Path.GetExtension(iterator.CurrentFileName());
+                        string currentExtension = Path.GetExtension(iterator.CurrentFileName()).ToLowerInvariant();
```
[the remark this answers is not in the document]
Comment (kc284) [reply XenModel/Actions/Updates/DownloadAndUnzipXenServerPatchAction.cs:203]: The bit `.ToLowerInvariant()` still needs to move from currentExtension in line 203 to currentExtension in line 205.

File: XenModel/Actions/Updates/DownloadAndUnzipXenServerPatchAction.cs [hunk 3]
```diff
@@ -80,8 +80,8 @@ public class DownloadAndUnzipXenServerPatchAction : AsyncAction, IByteProgressAc
             updateName = patchName;
             address = uri;
             downloadUpdate = address != null;
-            updateFileSuffixes = (from item in updateFileExtensions select '.' + item).ToArray();
-            skipUnzipping = downloadUpdate && updateFileSuffixes.Any(item => address.ToString().Contains(item));
+            updateFileSuffixes = (from item in updateFileExtensions select $".{item}").ToArray();
+            skipUnzipping = downloadUpdate && updateFileSuffixes.Any(item => address.ToString().ToLowerInvariant().Contains(item));
```
Comment (kc284) [inline XenModel/Actions/Updates/DownloadAndUnzipXenServerPatchAction.cs:84]: The item in the Contains still needs ToLowerInvariant().

**Review by kc284 — APPROVED** [review]

**Pull request merged**